!include CommonSetup

''With this test, we show you how to test for expected errors.''

So, we create a test, as usual.

!|Table:CreateTask|
| fixturename | title     | start_date | end_date | note |
| Demo fixture| Nice Task | 08:30 AM   | 10:30 AM | Nice note |

And then we go and try to change its start date to a weird value: It is supposed to start after it ends.
This is not supposed to happen, so we expect an error.

!|Table:UpdateTask|
| fixturename | start_date | expectedError |
| Demo fixture| 11:00 AM   | TSK-0040 |

The difference to our previous tests is that here, we do not use a ''VerifyTask'' to check for expected errors, like we would for expected values: Since errors occur the second the are sent to the server, they have to be checked during the step in which they are raised.

In the last case, this was during the update. If you try to create the task with broken values, you will have to catch the error during creation, of course:

!|Table:CreateTask|
| fixturename     | title     | start_date | end_date | expectedError |
| Another fixture | Nice Task | 11:30 AM   | 10:30 AM | TSK-0040      |

